### DISCUSSION TITLE
Preconditioners in Moose

### DISCUSSION THREAD

**Discussion by nanguaxiaofendui** [idaholab/moose#23635] [Q&A General] [answered]
Hello all,
I'd like to ask you about the difference between SMP,FSP and PBP. Can you explain how to choose these preprocessors? Are there any references or specific examples? Thank you very much.

**Comment by GiudGiud** [comment, marked as answer]
Hello

Please refer to their respective documentation
https://mooseframework.inl.gov/syntax/Preconditioning/index.html

You need to search the litterature for the postprocessing that works for your problem then reproduce it with one of the MOOSE ones. 
For example, for the Navier Stokes equations, a field split between pressure and velocity is a published solution we have tried in MOOSE NS

Guillaume

**Reply by nanguaxiaofendui** [reply]
thank you very much
